### PR TITLE
Builded a database

### PR DESCRIPTION
**Pull request by AnnaLangner**

### CODE REVIEW

File: script/main.py [hunk 1]
```diff
@@ -1,5 +1,12 @@
 #A script that takes command and parameter as an argument
 import argparse
+import json 
+import sqlite3
+from sqlite3 import Error
```
Comment (klangner) [inline script/main.py]: This is not necessary. Check how to deal with it in the code

File: script/main.py [hunk 2]
```diff
@@ -20,9 +27,56 @@ def init_argparser():
 
 
 def init_db():  
-  '''Initializing the database'''
+  '''Initializing the database'''  
+    
+  conn = None
+  try:
+    conn = sqlite3.connect("db/db_file.db")
+  except Error as e:
```
Comment (klangner) [inline script/main.py]: You can change `Error` to `sqlite3.Error`

**Review by klangner — CHANGES_REQUESTED** [review]

**Review by klangner — APPROVED** [review]

File: script/main.py [hunk 3]
```diff
@@ -19,10 +104,120 @@ def init_argparser():
...
+
+def insert_users_to_table(conn, users):
+  sql = '''INSERT INTO users VALUES (?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?)'''
+  cur = conn.cursor()
+  for columns in people:
+    cur.execute(sql, columns)  
+  conn.commit()
+
+
 def init_db():
```
Comment (klangner) [inline script/main.py]: split this function into initialization and adding data (import users)

File: script/main.py [hunk 4]
```diff
@@ -1,5 +1,90 @@
...
 import argparse
+import json 
+import sqlite3
+
+
+JSON_FILE = "init\persons.json"
+unflat_people =  json.load(open(JSON_FILE, encoding='utf-8'))
+
+
+people = []
```
Comment (klangner) [inline script/main.py]: do not use global variables here

File: script/main.py [hunk 5]
```diff
@@ -19,10 +104,120 @@ def init_argparser():
...
+    id_name text,
+    id_value text,
+    picture_large text,
+    picture_medium text,
+    picture_thumbnail text,
+    nat
+  ); '''
 
-  print('Nothing is implemented')
+  conn = create_connection(database)
```
Comment (klangner) [inline script/main.py]: do not create connection here. connection to the database should be passed as an argument.

**Review by klangner — CHANGES_REQUESTED** [review]

File: script/main.py [hunk 6]
```diff
@@ -19,10 +25,160 @@ def init_argparser():
...
+  conn = None
+  try:
+    conn = sqlite3.connect(db_file)
+  except sqlite3.Error as e:
+    print(e)
+  
+  return conn
+
+
+def create_table(conn, create_table_sql):
```
Comment (klangner) [inline script/main.py]: It would be better to move SQL text here and call this function create_user_table(conn)

File: script/main.py [hunk 7]
```diff
@@ -19,10 +25,160 @@ def init_argparser():
...
+  if conn is not None:
+    create_table(conn, sql_create_users_table)
+  else:
+    print("Error! cannot create the database connection.")
+  
+
+def import_users_to_db(conn):
+  people = []
+
+  list_of_people = unflat_people['results']
```
Comment (klangner) [inline script/main.py]: unflat_people (why this name?) should be passed as an argument here not as global variable

**Review by klangner — CHANGES_REQUESTED** [review]

File: script/main.py [hunk 8]
```diff
@@ -19,10 +21,159 @@ def init_argparser():
...
+  '''Initializing the database'''    
+
+  # create table
+  if conn is not None:
+    create_users_table(conn)
+  else:
+    print("Error! cannot create the database connection.")
+  
+
+def import_users_to_db(conn, unflat_people):
```
Comment (klangner) [inline script/main.py]: Rename parameter `unflattering_people` to something more meaningful. What is inside this parameter?

**Review by klangner — CHANGES_REQUESTED** [review]

File: script/main.py (hunk 8 above)
Comment (AnnaLangner) [reply script/main.py]: list that includes dictionaries

**Review by klangner — APPROVED** [review]

**Pull request merged**